neural_modelling/src/neuron/plasticity/stdp/weight_dependence/weight_additive_one_term_impl.c

```diff
@@ -36,27 +45,26 @@ address_t weight_initialise(
 
     // Copy plasticity region data from address
     // **NOTE** this seems somewhat safer than relying on sizeof
-    int32_t *plasticity_word = (int32_t *) address;
-    plasticity_weight_region_data =
+    additive_one_term_config_t *config = (additive_one_term_config_t *) address;
+
+    plasticity_weight_region_data_t *dtcm_copy = plasticity_weight_region_data =
             spin1_malloc(sizeof(plasticity_weight_region_data_t) * n_synapse_types);
-    if (plasticity_weight_region_data == NULL) {
+    if (dtcm_copy == NULL) {
         log_error("Could not initialise weight region data");
         return NULL;
     }
-    for (uint32_t s = 0; s < n_synapse_types; s++) {
-        plasticity_weight_region_data[s].min_weight = *plasticity_word++;
-        plasticity_weight_region_data[s].max_weight = *plasticity_word++;
-        plasticity_weight_region_data[s].a2_plus = *plasticity_word++;
-        plasticity_weight_region_data[s].a2_minus = *plasticity_word++;
+    for (uint32_t s = 0; s < n_synapse_types; s++, config++) {
+        dtcm_copy[s].min_weight = config->min_weight;
+        dtcm_copy[s].max_weight = config->max_weight;
+        dtcm_copy[s].a2_plus = config->a2_plus;
+        dtcm_copy[s].a2_minus = config->a2_minus;
 
         log_debug("\tSynapse type %u: Min weight:%d, Max weight:%d, A2+:%d, A2-:%d",
-                s, plasticity_weight_region_data[s].min_weight,
-                plasticity_weight_region_data[s].max_weight,
-                plasticity_weight_region_data[s].a2_plus,
-                plasticity_weight_region_data[s].a2_minus);
+                s, dtcm_copy[s].min_weight, dtcm_copy[s].max_weight,
+                dtcm_copy[s].a2_plus, dtcm_copy[s].a2_minus);
     }
     log_debug("weight_initialise: completed successfully");
 
     // Return end address of region
-    return (address_t) plasticity_word;
+    return (address_t) config;
```

Comment:
As with a previous comment, I think this highlights the error in the master code that this doesn't return where it has been read up to.  I suspect this is less of an issue here due to the fact that this is normally called last, but maybe it should do what it says!

Reply:
This returns the address immediately after where the last bit of configuration was read from. Note that we do `config++` in the loop, _including in the final step._

Reply:
Ah good again, missed the ++ bit.